root.c

```diff
@@ -902,6 +902,35 @@ static void stat_path(w_root_t *root,
 
     memcpy(&file->stat, &st, sizeof(file->stat));
 
+#ifndef _WIN32
+    // check for symbolic link
+    if (S_ISLNK(st.mode)) {
+      char link_target_path[WATCHMAN_NAME_MAX];
+      ssize_t tlen = 0;
+
```

Comment:
let's put this bit up here:

```
 if (file->symlink_target) {
       w_string_delref(file->symlink_target);
       file->symlink_target = NULL;
 }
```

so that we don't need to have three other places that we do the same thing; we can just focus on setting it in the success case.